File: namsor/java/src/main/java/oseg/namsor_examples/ChineseNameCandidatesGenderBatchExample.java
package oseg.namsor_examples;

import app.namsor.client.ApiException;
import app.namsor.client.Configuration;
import app.namsor.client.api.*;
import app.namsor.client.auth.*;
import app.namsor.client.JSON;
import app.namsor.client.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class ChineseNameCandidatesGenderBatchExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("api_key")).setApiKey("YOUR_API_KEY");

        var personalNames1 = new FirstLastNameGenderIn();
        personalNames1.id("e630dda5-13b3-42c5-8f1d-648aa8a21c42");
        personalNames1.firstName("LiYing");
        personalNames1.lastName("Zhao");
        personalNames1.gender("female");

        var personalNames = new ArrayList<FirstLastNameGenderIn>(List.of (
            personalNames1
        ));

        var batchFirstLastNameGenderIn = new BatchFirstLastNameGenderIn();
        batchFirstLastNameGenderIn.personalNames(personalNames);

        try
        {
            var response = new ChineseApi(config).chineseNameCandidatesGenderBatch(
                batchFirstLastNameGenderIn
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling ChineseApi#chineseNameCandidatesGenderBatch");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: namsor/kotlin/src/main/kotlin/oseg/namsor_examples/ChineseNameCandidatesGenderBatchExample.kt
package oseg.namsor_examples

import app.namsor.client.infrastructure.*
import app.namsor.client.apis.*
import app.namsor.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class ChineseNameCandidatesGenderBatchExample
{
    fun chineseNameCandidatesGenderBatch()
    {
        ApiClient.apiKey["X-API-KEY"] = "YOUR_API_KEY"

        val personalNames1 = FirstLastNameGenderIn(
            id = "e630dda5-13b3-42c5-8f1d-648aa8a21c42",
            firstName = "LiYing",
            lastName = "Zhao",
            gender = "female",
        )

        val personalNames = arrayListOf<FirstLastNameGenderIn>(
            personalNames1,
        )

        val batchFirstLastNameGenderIn = BatchFirstLastNameGenderIn(
            personalNames = personalNames,
        )

        try
        {
            val response = ChineseApi().chineseNameCandidatesGenderBatch(
                batchFirstLastNameGenderIn = batchFirstLastNameGenderIn,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling ChineseApi#chineseNameCandidatesGenderBatch")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling ChineseApi#chineseNameCandidatesGenderBatch")
            e.printStackTrace()
        }
    }
}
